src/include/brt.h

```diff
@@ -0,0 +1,265 @@
+ * if the number of modified block in a chunk are less and a bitmap representtaion if nearly all the blocks are modified
+ *
+ * In array representation, we don't need to store the entire block number instead we store each block number as a 2-byte
+ * offset from the start of the chunk.
+ *
+ * These same basic representational choices are used both when a block reference table is stored in memory
+ * and when it is serialized to disk.
+ *
+ */
+#define BITS_PER_BYTE               8
+#define BLOCKS_PER_CHUNK          (1 << 16)
+#define BLOCKS_PER_ENTRY          (BITS_PER_BYTE * sizeof(uint16_t))
+#define MAX_ENTRIES_PER_CHUNK     (BLOCKS_PER_CHUNK / BLOCKS_PER_ENTRY)
+#define INITIAL_ENTRIES_PER_CHUNK   16
+#define  BLOCKS_PER_READ             512
+/* Magic number for serialization file format. */
+#define BLOCKREFTABLE_MAGIC         0x652b137b
+
+typedef uint16_t* block_ref_table_chunk;
+
+/**
+ * A block reference table monitors and records the state of each fork separately.
+ * The key is used to search for the block entry in the ART
+ */
+typedef struct block_ref_table_key
+{
+   struct rel_file_locator rlocator; /**< The relation file locator for the relation fork */
+   enum fork_number forknum;      /**< The fork number of the relation fork */
+} block_ref_table_key;
+
+/**
+ * State for one relation fork.
+ *
+ * 'rlocator' and 'forknum' identify the relation fork to which this entry
+ * pertains.
+ *
+ * 'limit_block' represents the smallest known size (in blocks) of a relation during the range of LSNs (log sequence numbers) that a specific block reference table covers.
+ * - If the relation fork is either created or dropped, this value should be set to 0.
+ * - If the relation is truncated, it should be set to the number of blocks remaining after the truncation.
+ *
+ * 'nchunks' is the allocated length of each of the three arrays that follow. We can only represent the
+ * status of block numbers less than nchunks * BLOCKS_PER_CHUNK.
+ *
+ * 'chunk_size' is an array storing the allocated size of each chunk.
+ *
+ * 'chunk_usage' is an array storing the number of elements used in each chunk. If that value is less
+ * than MAX_ENTRIES_PER_CHUNK, the corresponding chunk is used as an array; else the corresponding
+ * chunk is used as a bitmap. When used as a bitmap, the least significant bit of the first array element
+ * is the status of the lowest-numbered block covered by this chunk.
+ *
+ * 'chunk_data' is the array of chunks.
+ */
+typedef struct block_ref_table_entry
+{
+   block_ref_table_key key;     /**< The key used to search for the block entry in the ART */
+   block_number limit_block;          /**< The limit block for the relation fork */
+   uint32_t nchunks;          /**< The number of chunks for the relation fork */
+   uint16_t* chunk_size;           /**< The size of each chunk in the relation fork */
+   uint16_t* chunk_usage;           /**< The number of used entries in each chunk */
```

Comment:
This only means something when data is in array representation right? You can make it clearer.